### PR TITLE
Rewrite developer guide

### PR DESCRIPTION
**Pull request by hairmare**
It's now more of an introduction to our development infra rather than a disfunctional normativy non-normative text.

The latter parts that are still written in RFC 2119 style still need refactoring and I'm thinking about
adding more docs on the vagrant setup.

I might also start writing up my new docker setup that almost makes sense locally. It's rather close to how we deploy but still needs some finishing touches and tests.

I'm opening this as a PR now since I'd like to gauge some interest. What would be more appealing to folks? More on the (might i add badly-maintained) [vagrant setup](https://github.com/gravity-platform/vagrant-centos-7-php) or more on a new docker based approach that has been ripening on my SSDs for sometime now.

If the latter has any demand I might as well start consolidating my >9000 repos on that and finish it before doing/redoing that part of this doc.

I've WIPed this PR for now, feel free to push into it as will as long as you stay on topic with the general spirit of the file. Maybe the following tasks inspire someone to put some time into this.

* [x] rewrite remaining RFC 2119 sections
* [x] add heaps more links to ALL THE THINGS
* [X] ~~add~~ remove graviton stuff
* [x] ~~add frontend & worker stuff~~
* [x] add vagrant box setup instructions
* [x] proofread

### CODE REVIEW

File: source/_pages/guides/developer_guide.md [hunk 1]
```diff
@@ -2,143 +2,118 @@
...
 
-## Code Repositories
+Our gravity-platform services are written in a few different languages and are organized
+into the following tiers.
 
-The gravity-platform SHALL be hosted on the following github organisations.
+- [mongodb](https://www.mongodb.org/) as data tier
+- graviton, a [symfony](https://symfony.com/) based JSON/REST server generation toolkit as api tier
+- several workers as logic tier
+- various angular2 or mobile native presentation tiers that stay in-house for the most part
```
Comment (wegenmic) [inline source/_pages/guides/developer_guide.md]: should probably change to a more generic term, like

> various presentation tiers that communicate with the graviton api

**Comment by wegenmic** [comment]
Since we try to get rid of the RFC 2119 style, we should it also avoid on new files (e.g. service_generator_guide.md)

**Comment by wegenmic** [comment]
:+1:

**Comment by hairmare** [comment]
@wegenmic `service_generator_guide.md` is just a copy paste of the old contents from `developer_guide.md` to get that one finished... I'll revisit that later (possibly with #24).

**Pull request merged**